src/agent/block_store_services/block_store_fs.js

```diff
@@ -230,22 +254,39 @@ class BlockStoreFs extends BlockStoreBase {
                 usage.count -= 1;
 
                 // also make sure we do not leave old .meta files on overwrite
-                const meta_path = this._get_block_meta_path(block_md.id);
                 const overwrite_meta_stat = await nb_native().fs.stat(fs_context, meta_path).catch(ignore_not_found);
                 if (overwrite_meta_stat) {
                     usage.size -= overwrite_meta_stat.size;
-                    await nb_native().fs.unlink(fs_context, meta_path).catch(ignore_not_found);
+                    if (!this.xattr_fallback) {
```

Comment:
I think this check is undesired - we already found the meta file, so we should delete it unconditionally.